pandas/core/indexes/category.py

```diff
@@ -696,8 +696,11 @@ def get_indexer_non_unique(self, target):
 
     @Appender(_index_shared_docs["_convert_scalar_indexer"])
     def _convert_scalar_indexer(self, key, kind=None):
-        if self.categories._defer_to_indexing:
-            return self.categories._convert_scalar_indexer(key, kind=kind)
+        if kind == "loc" or self.categories._defer_to_indexing:
+            try:
+                return self.categories._convert_scalar_indexer(key, kind=kind)
+            except TypeError:
+                self._invalid_indexer("label", key=key)
```

Comment:
when is this TypeError hit?

Reply:
That was hit when you have ``pd.Series([1,2], index=pd.CategoricalIndex(["a", "b"])).loc[1]``, i.e. wrong indexing type. It's the same error type as in ```s = pd.Series([1,2], index=["a", "b"]).loc[1]``

I've added a test for it.